### ISSUE TITLE
Advanced Filtering with Dynamic LINQ

### ISSUE THREAD

**Issue by tobimax** [TabularEditor/TabularEditor3#59]
Hi,

I was looking for the advanced filtering and found it with #27. In v2 you would get a list of filtered objects that you could then bulk edit. This is no longer possible, it's a great feature and feels like a step backward. 

Would it be possible to please re-introduce Dynamic LINQ in the text search area as well?

Regards

Toby

**Comment by otykier** [comment]
I deliberately removed Dynamic LINQ from the TOM explorer since there was some confusion as to how the search results were displayed in the open-source version of Tabular Editor. However, I fully understand the need to be able to bulk edit the search results.

I propose one of the following alternative:

- Selecting or multi-selecting objects in the Find Results view will display the selected objects in the Property Grid, so you can still edit in bulk.
- Alternatively, a button in the Find Results view that will select the (selected) search results in the TOM Explorer to keep the UI consistent (Property Grid should always follow the selection in the TOM Explorer).

Before going there, however, I'd like to understand the requirement in more details. Are you aware that you can filter columns by object type in the TOM Explorer in combination with a text search?

![image](https://user-images.githubusercontent.com/8976200/121310999-433c2780-c904-11eb-9af7-6b873bdf9a30.png)

**Comment by tobimax** [comment]
Yes, I can see the Object Type Filter, that is useful. I think I preferred TE2 method of filtering the TOM Explorer, however a button in the Find Results view that will select the (selected) search results in the TOM Explorer would also work. On big models it might look a bit unwieldy in the TOM Explorer unless other objects were hidden.

I tend to use the LINQ Search to look for invalid state, move common measures under folders, change data types or formatting etc. search for measures with certain content.

Thanks

Toby.

**Comment by otykier** [comment]
@tobimax just a small update. With [3.0.7](https://github.com/TabularEditor/TabularEditor3/releases/tag/3.0.7) the Object Type Filter now displays all object types and it actually works (there was a bug in the previous version, where the filter did nothing...)
